lib/knock/client.rb

```diff
@@ -18,7 +18,11 @@ def execute_request(request:)
       http_status = response.code.to_i
       handle_error_response(response: response) if http_status >= 400
 
-      JSON.parse(response.body)
+      if response.body and response.body != "" do
+        JSON.parse(response.body) 
+      else
+        nil
```

Comment:
nit: no need for the else here, an `if` statement returns nil if condition doesn't match and there is no else statement